app/views/products/show.html.erb

```diff
@@ -9,24 +7,27 @@
       <h3><b><%= @product.name %></b></h3>
       <h4><b>Seller: <%= link_to @product.merchant.username, merchant_path(@product.merchant.id) %></b></h4>
       <p><%= @product.description %></p>
-      <p><b><%=  "$#{'%.2f' % @product.price}" %></b></p>
+      <p><b><%= number_to_currency(@product.price) %></b></p>
+    
       <% if @product.inventory > 0 %>
         <p>In Stock</p>
       <% else %>
         <p>Out of Stock</p>
       <%end%>
-      <% if @login_merchant %>
-        <% status = (@product.active ? "Retire Product" : "Reactivate Product") %>
-        <%= button_to status, toggle_active_path(@product.id), class: "btn btn-danger", data: { confirm: "Are you sure you'd like to #{status.downcase}?"}, method: :patch %>
-      <% end %>
-
-      <%= button_to "delete", product_path(@product.id), class: "btn btn-danger", data: { confirm: "Are you sure you'd like to delete this product?"}, method: :delete %>
+      <% current_state = (@product.active ? "Retire Product" : "Reactivate Product") %>
+      <% if @login_merchant == @product.merchant %>
+        <%= button_to current_state, toggle_active_path(@product.id), class: "btn btn-warning", data: { confirm: "Are you sure you'd like to #{current_state.downcase}?"}, method: :patch %>
 
+        <%= button_to "delete", product_path(@product.id), class: "btn btn-danger", data: { confirm: "Are you sure you'd like to delete this product?"}, method: :delete %>
+      <% end%>
     </div>
```

Comment:
Love this! Thank you! I've also been thinking: should the toggle active and delete product buttons live in the Merchant Dashboard instead of products#show?

Reply:
That's a great question! I was wondering that myself, like it would make things more centralized and easier for the merchant if it was in the dashboard. I'm inclined to agree that it should be in the dashboard!